src/codecs/encoding/encoder.rs

```diff
@@ -7,7 +7,7 @@ use tokio_util::codec::Encoder as _;
 
 use crate::{
     event::Event,
-    internal_events::{EncoderFramingFailed, EncoderSerializeFailed},
+    internal_events::{EncoderFramingError, EncoderSerializeError},
```

Comment:
Interesting, wondering why we have two conflicting rules

> - `Verb` - the past tense verb describing when the event occurred (e.g., `Received`, `Sent`, `Processes`)
> - `[Error]` - if the event is an error it MUST end with `Error`

instead of going with e.g. `Failed` for errors